guides/v2.2/frontend-dev-guide/layouts/xml-manage.md

```diff
@@ -383,7 +382,95 @@ In this file, reference the element having added the `remove` attribute:
+
+**Example 2**: add functionality with a ViewModel inserted into the `cart/item/default.phtml` template found in `Magento/Checkout/view/frontend/layout/checkout_cart_item_renderers.xml`.
+
+Let us say that we want to add functionality to a core template with custom logic using a ViewModel in the `cart/item/default.phtml` template found in `Magento/Checkout/view/frontend/layout/checkout_cart_item_renderers.xml`:
+
+```xml
+<?xml version="1.0"?>
+<page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd">
+<body>
+    <referenceBlock name="Magento/Checkout/view/frontend/layout/checkout_cart_item_renderers.xml" template="Magento_Checkout::cart/item/default.phtml" />
+        <arguments>
+           <argument name="viewModel" xsi:type="object">Vendor\CustomModule\ViewModel\Class</argument>
+        </arguments>
+    </referenceBlock>
+</body>
+```
+
+Even though the scope for the template path is based on the block class, changing the block's class with a class preference in your `di.xml` file could lead to problems when overriding your template in a theme. To prevent this, the best way to change the template's module scope is to include a namespace in the template name, similar to how the core does in `checkout_cart_item_renderers.xml` (i.e. `Magento_Checkout::cart/item/default.phtml`).
+
+You would also have to implement the right interface in your viewModel class (i.e. `ArgumentInterface`):
+
+```php
+namespace Vendor\CustomModule\ViewModel;
+
+class Class implements \Magento\Framework\View\Element\Block\ArgumentInterface
+{
+    public function __construct()
+    {
+
+    }
+}
+```
+
+To substitute or extend the behavior of original, public methods for any class or interface, we can make use of plugins, or interceptors, which are classes that modify the behavior of public class functions by intercepting a function call and running code before, after, or around that function call. 
+
+This interception approach reduces conflicts among extensions that change the behavior of the same class or method, with a Plugin class implementation changing only the behavior of a class function, not the class itself.
+
+The di.xml file in your module declares a plugin for a class object in the following manner:
+
+```xml
+<config>
+    <type name="{ObservedType}">
+      <plugin name="{pluginName}" type="{PluginClassName}" sortOrder="1" disabled="false" />
+    </type>
+</config>
+```
+
+You must specify the following elements: type name (a class or interface which the plugin observes), plugin name (an arbitrary plugin name that identifies a plugin that is also used to merge the configurations for the plugin), and plugin type (name of a plugin’s class or its virtual type; use the following naming convention when you specify this element: \Vendor\Module\Plugin\<ClassName>).
```

Comment:
This seems to be duplicating content that should be on the actual Plugin / Interceptor topic.